src/components/UnifiedReport/download/useUnifiedReportDownload.ts

```diff
@@ -0,0 +1,140 @@
+import type { Key } from 'swr'
+import useSWRMutation, { type SWRMutationResponse } from 'swr/mutation'
+import { Schema } from 'effect'
+import { ReportEnum } from '../../../schemas/reports/unifiedReport'
+import {
+  useUnifiedReportWithDateParams,
+  type UnifiedReportWithDateParams,
+} from '../../../providers/UnifiedReportStore/UnifiedReportStoreProvider'
+import { useLayerContext } from '../../../contexts/LayerContext'
+import { useAuth } from '../../../hooks/useAuth'
+import { getBalanceSheetExcel } from '../../../api/layer/balance_sheet'
+import { getCashflowStatementCSV } from '../../../api/layer/statement-of-cash-flow'
+import { unsafeAssertUnreachable } from '../../../utils/switch/assertUnreachable'
+import { S3PresignedUrlSchema, S3PresignedUrlSchemaType } from '../../../schemas/common/s3PresignedUrl'
+
+type BalanceParams = Extract<UnifiedReportWithDateParams, { report: ReportEnum.BalanceSheet }>
+type CashflowParams = Extract<UnifiedReportWithDateParams, { report: ReportEnum.CashflowStatement }>
+
+const getTag = (report: ReportEnum) => `#download-unified-${report}-report`
+
+const DownloadUnifiedReportReturnSchema = Schema.Struct({
+  data: S3PresignedUrlSchema,
+})
+
+type DownloadUnifiedReportSWRMutationResponse =
+  SWRMutationResponse<S3PresignedUrlSchemaType, unknown, Key, never>
+
+class DownloadUnifiedReportSWRResponse {
+  private swrResponse: DownloadUnifiedReportSWRMutationResponse
+
+  constructor(swrResponse: DownloadUnifiedReportSWRMutationResponse) {
+    this.swrResponse = swrResponse
+  }
+
+  get data() {
+    return this.swrResponse.data
+  }
+
+  get trigger() {
+    return this.swrResponse.trigger
+  }
+
+  get isMutating() {
+    return this.swrResponse.isMutating
+  }
+
+  get isError() {
+    return this.swrResponse.error !== undefined
+  }
+}
+
+function buildKey({
+  access_token: accessToken,
+  apiUrl,
+  businessId,
+  reportWithDateParams,
+}: {
+  access_token?: string
+  apiUrl?: string
+  businessId: string
+  reportWithDateParams: UnifiedReportWithDateParams
+}) {
+  if (accessToken && apiUrl) {
+    return {
+      accessToken,
+      apiUrl,
+      businessId,
+      ...reportWithDateParams,
+      tags: [getTag(reportWithDateParams.report)],
+    }
+  }
+}
+
+const reportDownloadConfig = {
```

Comment:
I recommend we define a type for this config to assist our IDEs and future devs to ensure they conform to an explicitly defined shape.